reconcile/external_resources/model.py

```diff
@@ -289,15 +300,37 @@ def resolve_configuration(
                 module_type=None,
                 image=None,
                 version=None,
+                channel=None,
                 reconcile_timeout_minutes=None,
                 outputs_secret_image=None,
                 outputs_secret_version=None,
                 resources=None,
             )
 
+        provisioner_channel = (spec.provisioner.get("external_resources") or {}).get(
+            "channel"
+        ) or None
+
+        channel_name = (
+            module_overrides.channel or provisioner_channel or module.default_channel
+        )
+
+        channel_data = next(
+            ((c.image, c.version) for c in module.channels if c.name == channel_name),
+            None,
+        )
+
+        if channel_data is None:
+            raise ExternalResourceModuleConfigurationError(
+                f"Module {module.provision_provider}/{module.provider} does not have the channel {channel_name}. "
+                f"Resource spec: {spec.id_object()}"
+            )
+
+        channel_image, channel_version = channel_data
```

Comment:
is pack and unpack needed here as we can just find channel first, then use fields like this

```python
        channel = next(
            (c for c in module.channels if c.name == channel_name),
            None,
        )
        ...
        return ExternalResourceModuleConfiguration(
            image=module_overrides.image or channel.image,
            version=module_overrides.version or channel.version,
```

Reply:
Fixed.